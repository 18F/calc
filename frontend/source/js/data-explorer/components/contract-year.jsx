import React from 'react';
import { connect } from 'react-redux';

import Tooltip from './tooltip';
import { setContractYear as setContractYearAction } from '../actions';
import {
  CONTRACT_YEAR_CURRENT,
  CONTRACT_YEAR_1,
  CONTRACT_YEAR_2,
  CONTRACT_YEAR_LABELS,
} from '../constants';

function ContractYear({ idPrefix, contractYear, setContractYear }) {
  const idCurrent = `${idPrefix}current-year`;
  const id1 = `${idPrefix}one-year-out`;
  const id2 = `${idPrefix}two-years-out`;
  const setYear = year => () => { setContractYear(year); };

  return (
    <div className="filter contract-year">
      <fieldset className="fieldset-inputs">

        <legend>Contract year:</legend>
        <a href="#" className="tooltip filter-more-info">
           <Tooltip text="All five years of pricing are available in the export">
             What's this?
           </Tooltip>
        </a>
        <h3 className="sr-only">Contract Year</h3>

        <ul className="contract-year-block">
          <li className="contract-list-item">
            <label htmlFor={idCurrent} className="radio">
              <input id={idCurrent} type="radio"
                     checked={contractYear === CONTRACT_YEAR_CURRENT}
                     onChange={setYear(CONTRACT_YEAR_CURRENT)}
                     name="contract-year" value={CONTRACT_YEAR_CURRENT}
                     tabIndex="0" />
              <span tabIndex="-1"
                    className="contract-current-year-span checkbox-focus"
                    aria-hidden="true">Current</span>
              <span className="sr-only">
                {CONTRACT_YEAR_LABELS[CONTRACT_YEAR_CURRENT]}
              </span>
            </label>
          </li>
          <li className="contract-list-item">
            <label htmlFor={id1} className="radio">
              <input id={id1} type="radio"
                     checked={contractYear === CONTRACT_YEAR_1}
                     onChange={setYear(CONTRACT_YEAR_1)}
                     name="contract-year" value={CONTRACT_YEAR_1}
                     tabIndex="0" />
              <span tabIndex="-1"
                    className="checkbox-focus"
                    aria-hidden="true">+1</span>
              <span className="sr-only">
                {CONTRACT_YEAR_LABELS[CONTRACT_YEAR_1]}
              </span>
            </label>
          </li>
          <li className="contract-list-item">
            <label htmlFor={id2} className="radio">
              <input id={id2} type="radio"
                     name="contract-year" value={CONTRACT_YEAR_2}
                     checked={contractYear === CONTRACT_YEAR_2}
                     onChange={setYear(CONTRACT_YEAR_2)}
                     tabIndex="0" />
              <span tabIndex="-1"
                    className="contract-last-year-span checkbox-focus"
                    aria-hidden="true">+2</span>
              <span className="sr-only">
                {CONTRACT_YEAR_LABELS[CONTRACT_YEAR_2]}
              </span>
            </label>
          </li>
        </ul>
      </fieldset>
    </div>
  );
}

ContractYear.propTypes = {
  contractYear: React.PropTypes.string.isRequired,
  setContractYear: React.PropTypes.func.isRequired,
  idPrefix: React.PropTypes.string,
};

ContractYear.defaultProps = {
  idPrefix: '',
};

export default connect(
  state => ({ contractYear: state['contract-year'] }),
  { setContractYear: setContractYearAction }
)(ContractYear);
