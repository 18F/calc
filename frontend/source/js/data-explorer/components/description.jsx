import React from 'react';
import { connect } from 'react-redux';

import {
  MIN_EXPERIENCE,
  MAX_EXPERIENCE,
  EDU_LABELS,
  SITE_LABELS,
  BUSINESS_SIZE_LABELS,
  SCHEDULE_LABELS,
} from '../constants';

import { formatCommas } from '../util';

function Filter({
  extraClassName,
  label,
  children,
}) {
  let className = 'filter';

  if (extraClassName) {
    className += ` ${extraClassName}`;
  }

  return (
    <span className={className}>
      {label ? `${label}: ` : null}
      <a className="focus-input" href="#">
        {children}
      </a>
    </span>
  );
}

Filter.propTypes = {
  extraClassName: React.PropTypes.string,
  label: React.PropTypes.string,
  children: React.PropTypes.any.isRequired,
};

function Description({
  shownResults,
  totalResults,
  minExperience,
  maxExperience,
  education,
  site,
  businessSize,
  schedule,
  laborCategory,
}) {
  let results = ' results ';
  const filtersClasses = ['filters'];
  const filters = [];

  if (laborCategory) {
    filters.push(<Filter key="lab">{laborCategory}</Filter>);
  }

  if (education.length) {
    filters.push(
      <Filter key="edu" label="education level"
              extraClassName="education-filter">
        {education.map(x => EDU_LABELS[x]).join(', ')}
      </Filter>
    );
  }

  if (minExperience !== MIN_EXPERIENCE ||
      maxExperience !== MAX_EXPERIENCE) {
    filters.push(
      <Filter key="exp" label="experience">
        {minExperience} - {maxExperience} years
      </Filter>
    );
  }

  if (site) {
    filters.push(
      <Filter key="sit" label="worksite">
        {SITE_LABELS[site]}
      </Filter>
    );
  }

  if (businessSize) {
    filters.push(
      <Filter key="bus" label="business size">
        {BUSINESS_SIZE_LABELS[businessSize]}
      </Filter>
    );
  }

  if (schedule) {
    filters.push(
      <Filter key="sch" label="schedule">
        {SCHEDULE_LABELS[schedule]}
      </Filter>
    );
  }

  if (filters.length) {
    results += 'with ';
  } else {
    filtersClasses.push('hidden');
  }

  // TODO: The original version of this faded-in (but never out)
  // whenever it changed. We might want to do that too, or choose
  // a different animation.

  // TODO: The original version of this didn't show the filters on
  // the very first request (it also kept track of how many requests
  // had been made) but I'm not sure why that was good for usability,
  // so I've left the functionality out for now. We can bring it back
  // if needed.

  return (
    <p className="">
      {`Showing ${formatCommas(shownResults)} of `}
      <span className="total">{formatCommas(totalResults)}</span>
      {results}
      <span className={filtersClasses.join(' ')}>
        {filters}
      </span>
    </p>
  );
}

Description.propTypes = {
  shownResults: React.PropTypes.number.isRequired,
  totalResults: React.PropTypes.number.isRequired,
  minExperience: React.PropTypes.number.isRequired,
  maxExperience: React.PropTypes.number.isRequired,
  education: React.PropTypes.array.isRequired,
  site: React.PropTypes.string.isRequired,
  businessSize: React.PropTypes.string.isRequired,
  schedule: React.PropTypes.string.isRequired,
  laborCategory: React.PropTypes.string.isRequired,
};

function mapStateToProps(state) {
  return {
    shownResults: state.rates.data.results.length,
    totalResults: state.rates.data.count,
    minExperience: state.min_experience,
    maxExperience: state.max_experience,
    education: state.education,
    site: state.site,
    businessSize: state.business_size,
    schedule: state.schedule,
    laborCategory: state.q,
  };
}

export default connect(mapStateToProps)(Description);
