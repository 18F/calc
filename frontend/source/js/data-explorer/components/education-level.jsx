/* global document */

import React from 'react';
import { connect } from 'react-redux';

import {
  autobind,
  handleEnterOrSpace,
} from '../util';

import {
  EDU_LABELS,
} from '../constants';

import {
  toggleEducationLevel,
} from '../actions';

// TODO: We could just use jQuery for this, but I wanted to decouple
// the new React code from jQuery as much as possible for now.
function elementContains(container, contained) {
  let target = contained.parentNode;

  while (target) {
    if (target === container) {
      return true;
    }
    target = target.parentNode;
  }
  return false;
}

/**
 * The following logic was created to mimic the following
 * legacy jQuery behavior:
 *
 *   Dropdown with Multiple checkbox select with jQuery - May 27, 2013
 *   (c) 2013 @ElmahdiMahmoud
 *   license: http://www.opensource.org/licenses/mit-license.php
 */

class EducationLevel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      expanded: false,
    };
    autobind(this, ['handleToggleMenu', 'handleDocumentClick']);
  }

  handleDocumentClick(e) {
    if (!this.state.expanded) {
      return;
    }

    // Whenever users click outside of our dropdown, hide it.
    if (!elementContains(this.dropdownEl, e.target)) {
      this.setState({ expanded: false });
    }
  }

  handleToggleMenu(e) {
    // TODO: The original jQuery logic used .slideDown() here, it'd
    // be nice if we could use some sort of transition too.

    e.preventDefault();
    this.setState({
      expanded: !this.state.expanded,
    });
  }

  handleCheckboxClick(level) {
    this.props.toggleEducationLevel(level);
  }

  componentDidMount() {
    document.addEventListener('click', this.handleDocumentClick);
    document.addEventListener('focus', this.handleDocumentClick, true);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.handleDocumentClick);
    document.removeEventListener('focus', this.handleDocumentClick, true);
  }

  render() {
    const levels = this.props.levels;
    const idPrefix = this.props.idPrefix;
    const ddStyle = this.state.expanded ? { display: 'block' } : null;
    const inputs = Object.keys(EDU_LABELS).map(value => {
      const id = idPrefix + value;

      return (
        <li key={value}>
          <input
            id={id}
            type="checkbox"
            value={value}
            checked={levels.indexOf(value) >= 0}
            onChange={this.handleCheckboxClick.bind(this, value)}
            name="education" />
          <label htmlFor={id}>{EDU_LABELS[value]}</label>
        </li>
      );
    });
    let linkContent;

    if (levels.length === 0) {
      linkContent = (
        <span className="eduSelect">Select
          <span className="sr-only"> to reveal Education Level options</span>
        </span>
      );
    } else {
      const selectedLevels = levels.map(value => {
        const label = EDU_LABELS[value];
        return <span key={value} title={label}>{label}</span>;
      });
      linkContent = (
        <div className="multiSel">{selectedLevels}</div>
      );
    }

    return (
      <div>
        <label>Education level:</label>
        <dl className="dropdown"
            ref={(el) => { this.dropdownEl = el; } }>
          <dt>
            <a href="#" onClick={this.handleToggleMenu}
               role="button"
               aria-expanded={this.state.expanded.toString()}
               onKeyDown={handleEnterOrSpace(this.handleToggleMenu)}
               className={levels.length === 0 ? '' : 'filter_active'}>
              {linkContent}
            </a>
          </dt>

          <dd>
            <div className="multiSelect">
              <fieldset>
                <legend className="sr-only">Education level:</legend>
                <ul style={ddStyle}>
                  {inputs}
                </ul>
              </fieldset>
            </div>
          </dd>
        </dl>
      </div>
    );
  }
}

EducationLevel.propTypes = {
  levels: React.PropTypes.array.isRequired,
  idPrefix: React.PropTypes.string,
  toggleEducationLevel: React.PropTypes.func.isRequired,
};

EducationLevel.defaultProps = {
  idPrefix: 'education-level-',
};

export default connect(
  state => ({ levels: state.education }),
  { toggleEducationLevel }
)(EducationLevel);
