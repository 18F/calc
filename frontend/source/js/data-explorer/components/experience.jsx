/* global $ */

import React from 'react';
import { connect } from 'react-redux';

import {
  setExperience,
} from '../actions';

import {
  autobind,
} from '../util';

import {
  MIN_EXPERIENCE,
  MAX_EXPERIENCE,
} from '../constants';

class Experience extends React.Component {
  constructor(props) {
    super(props);
    autobind(this, ['onSliderSlide', 'onSliderChange']);
    this.state = {
      isSliding: false,
      sliderVal: null,
    };
  }

  componentDidMount() {
    // We're currently using an old version of noUiSlider (7.0.10). Its
    // documentation can only be seen via archive.org:
    //
    // https://web.archive.org/web/20141224210103/http://refreshless.com/nouislider/

    $(this.sliderEl).noUiSlider({
      start: [
        this.props.min,
        this.props.max,
      ],
      step: 1,
      connect: true,
      range: {
        min: MIN_EXPERIENCE,
        max: MAX_EXPERIENCE,
      },
    });
    $(this.sliderEl).on({
      slide: this.onSliderSlide,
      change: this.onSliderChange,
    });
  }

  componentDidUpdate() {
    if (!this.state.isSliding) {
      const [min, max] = this.getSliderVal();

      if (min !== this.props.min || max !== this.props.max) {
        $(this.sliderEl).val([this.props.min, this.props.max]);
      }
    }
  }

  getSliderVal() {
    return $(this.sliderEl).val().map(x => parseInt(x, 10));
  }

  onSliderSlide() {
    $('.noUi-horizontal .noUi-handle', this.rootEl).addClass('filter_focus');

    this.setState({
      isSliding: true,
      sliderVal: this.getSliderVal(),
    });
  }

  onSliderChange() {
    $('.noUi-horizontal .noUi-handle', this.rootEl)
      .removeClass('filter_focus');

    const [min, max] = this.getSliderVal();

    this.setState({
      isSliding: false,
      sliderVal: null,
    });

    if (min !== this.props.min) {
      this.props.dispatch(setExperience('min', min));
    }
    if (max !== this.props.max) {
      this.props.dispatch(setExperience('max', max));
    }
  }

  makeOptions(min, max) {
    const options = [];

    for (let i = min; i <= max; i++) {
      options.push(<option key={i} value={i}>{i}</option>);
    }

    return options;
  }

  render() {
    let min = this.props.min;
    let max = this.props.max;

    if (this.state.isSliding) {
      [min, max] = this.state.sliderVal;
    }

    const minId = `${this.props.idPrefix}min`;
    const maxId = `${this.props.idPrefix}max`;
    const classes = ['select-small'];
    const minClasses = classes.concat(min === MIN_EXPERIENCE ? [] : [
      'filter_active',
    ]);
    const maxClasses = classes.concat(max === MAX_EXPERIENCE ? [] : [
      'filter_active',
    ]);
    const onChange = type => e => {
      this.props.dispatch(setExperience(type, parseInt(e.target.value, 10)));
    };

    return (
      <div className="filter" ref={(el) => { this.rootEl = el; }}>
        <fieldset>
          <legend>
            Experience:
          </legend>
          <div className="slider" ref={(el) => { this.sliderEl = el; }} />
          <div className="experience_range">
            <label htmlFor={minId} className="sr-only">Minimum Years</label>
            <select id={minId} value={min} name="min_experience"
                    onChange={onChange('min')}
                    className={minClasses.join(' ')}>
              {this.makeOptions(MIN_EXPERIENCE, max)}
            </select>
            {' - '}
            <label htmlFor={maxId} className="sr-only">Maximum Years</label>
            <select id={maxId} value={max} name="max_experience"
                    onChange={onChange('max')}
                    className={maxClasses.join(' ')}>
              {this.makeOptions(min, MAX_EXPERIENCE)}
            </select>
            {' years'}
          </div>
        </fieldset>
      </div>
    );
  }
}

Experience.propTypes = {
  min: React.PropTypes.number.isRequired,
  max: React.PropTypes.number.isRequired,
  dispatch: React.PropTypes.func.isRequired,
  idPrefix: React.PropTypes.string,
};

Experience.defaultProps = {
  idPrefix: 'experience-',
};

function mapStateToProps(state) {
  return {
    min: state.min_experience,
    max: state.max_experience,
  };
}

export default connect(mapStateToProps)(Experience);
