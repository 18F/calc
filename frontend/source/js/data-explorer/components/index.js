export { default as Description } from './description';
export { default as Highlights } from './highlights';
export { default as Histogram } from './histogram';
export { default as ProposedPrice } from './proposed-price';
export { default as EducationLevel } from './education-level';
export { default as Experience } from './experience';
export { default as ExportData } from './export-data';
export { default as ResultsTable } from './results-table';
export { default as Schedule } from './schedule';
export { default as ContractYear } from './contract-year';
export { default as QueryType } from './query-type';
export { default as Site } from './site';
export { default as BusinessSize } from './business-size';
