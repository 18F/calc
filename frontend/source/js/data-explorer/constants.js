export const MAX_EXPERIENCE = 45;
export const MIN_EXPERIENCE = 0;

export const HISTOGRAM_BINS = 12;

export const EDU_HIGH_SCHOOL = 'HS';
export const EDU_ASSOCIATES = 'AA';
export const EDU_BACHELORS = 'BA';
export const EDU_MASTERS = 'MA';
export const EDU_PHD = 'PHD';

export const EDU_LABELS = {
  [EDU_HIGH_SCHOOL]: 'High School',
  [EDU_ASSOCIATES]: 'Associates',
  [EDU_BACHELORS]: 'Bachelors Degree',
  [EDU_MASTERS]: 'Masters Degree',
  [EDU_PHD]: 'Ph.D',
};

export const BUSINESS_SIZE_LABELS = {
  s: 'small business',
  o: 'other than small',
};

export const SITE_LABELS = {
  customer: 'customer',
  contractor: 'contractor',
  both: 'both',
};

export const SCHEDULE_LABELS = {
  Consolidated: 'Consolidated',
  FABS: '520 - Legacy FABS',
  AIMS: '541 - Legacy AIMS',
  'Language Services': '73802 - Legacy Language',
  PES: '871 - Legacy PES',
  MOBIS: '874 - Legacy MOBIS',
  Logistics: '87405 - Legacy Logistics',
  Environmental: '899 - Legacy Environmental',
  'IT Schedule 70': '132 - IT 70',
};

export const CONTRACT_YEAR_LABELS = {
  current: 'Current year',
  1: 'One year out',
  2: 'Two years out',
};

export const DEFAULT_CONTRACT_YEAR = 'current';

export const EMPTY_RATES_DATA = {
  minimum: 0,
  maximum: 0.001,
  average: 0,
  first_standard_deviation: 0,
  count: 0,
  results: [],
  wage_histogram: [
    { count: 0, min: 0, max: 0 },
  ],
};
