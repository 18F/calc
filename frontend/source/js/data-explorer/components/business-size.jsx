import React from 'react';
import { connect } from 'react-redux';

import { makeOptions } from './util';
import { setBusinessSize as setBusinessSizeAction } from '../actions';
import { BUSINESS_SIZE_LABELS } from '../constants';

function BusinessSize({ idPrefix, size, setSize }) {
  const id = `${idPrefix}business_size`;
  const handleChange = e => { setSize(e.target.value); };

  return (
    <div className="filter filter-business_size">
      <label htmlFor={id}>Business size:</label>
      <select id={id} name="business_size"
              value={size} onChange={handleChange}>
        {makeOptions(BUSINESS_SIZE_LABELS)}
      </select>
    </div>
  );
}

BusinessSize.propTypes = {
  size: React.PropTypes.string.isRequired,
  setSize: React.PropTypes.func.isRequired,
  idPrefix: React.PropTypes.string,
};

BusinessSize.defaultProps = {
  idPrefix: '',
};

export default connect(
  state => ({ size: state.business_size }),
  { setSize: setBusinessSizeAction }
)(BusinessSize);
