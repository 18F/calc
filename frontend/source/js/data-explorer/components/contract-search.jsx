import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import { setSchedule as setScheduleAction } from '../actions';
import { setQueryBy as setQueryByAction } from '../actions';
import { QUERY_BY_CONTRACT} from '../constants';

export function ContractNum({ setSchedule, setQueryBy }) {
  const handleClick = (e) => {
    // unset selectedSchedule if it has been previously set
    setSchedule('');
    setQueryBy(e.target.value);
  }

  return (
    <button
      type="button"
      id={QUERY_BY_CONTRACT}
      name={QUERY_BY_CONTRACT}
      value={QUERY_BY_CONTRACT}
      onClick={handleClick}
    >
      Search by contract number
    </button>
  );
}

ContractNum.propTypes = {
  setQueryBy: PropTypes.func.isRequired,
};

export default connect(
  state => ({
    selectedSchedule: state.schedule,
    queryBy: state.query_by,
  }),
  {
    setSchedule: setScheduleAction,
    setQueryBy: setQueryByAction,
  },
)(ContractNum);
