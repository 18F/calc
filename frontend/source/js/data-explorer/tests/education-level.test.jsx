import toJson from 'enzyme-to-json';

import { EducationLevel } from '../components/education-level';
import makeSetup from './testSetup';
import { EDU_LABELS } from '../constants';

const defaultProps = {
  levels: ['MA', 'PHD'],
  idPrefix: 'zzz_',
  toggleEducationLevel: jest.fn(),
};

const setup = makeSetup(EducationLevel, defaultProps);

describe('<EducationLevel>', () => {
  it('renders correctly', () => {
    const { wrapper } = setup();

    Object.keys(EDU_LABELS).forEach((key) => {
      const checkbox = wrapper.find(`input[value="${key}"][type="checkbox"]`);
      expect(checkbox.exists()).toBeTruthy();
      const id = `zzz_${key}`;
      expect(checkbox.prop('id')).toBe(id);

      if (defaultProps.levels.indexOf(key) >= 0) {
        expect(checkbox.prop('checked')).toBeTruthy();
      } else {
        expect(checkbox.prop('checked')).toBeFalsy();
      }

      const label = wrapper.find(`label[htmlFor="${id}"]`);
      expect(label.exists()).toBeTruthy();
      expect(label.text()).toBe(EDU_LABELS[key]);
    });
  });

  it('matches snapshot', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has filter_active class when levels are selected', () => {
    let { wrapper } = setup({ levels: [] });
    expect(wrapper.find('a.filter_active').exists()).toBeFalsy();

    ({ wrapper } = setup({ levels: ['HS'] }));
    expect(wrapper.find('a.filter_active').exists()).toBeTruthy();
  });

  it('calls toggleEducationLevel when checkbox is clicked', () => {
    const { props, mounted } = setup();
    expect(props.toggleEducationLevel.mock.calls.length).toBe(0);

    mounted.find('input[value="HS"]')
      .simulate('change', { target: { checked: true } });

    expect(props.toggleEducationLevel.mock.calls.length).toBe(1);
    expect(props.toggleEducationLevel.mock.calls[0][0]).toBe('HS');
  });
});
