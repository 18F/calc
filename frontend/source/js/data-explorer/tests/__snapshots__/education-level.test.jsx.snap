exports[`<EducationLevel> matches snapshot 1`] = `
<div>
  <label>
    Education level:
  </label>
  <dl
    className="dropdown">
    <dt>
      <a
        aria-expanded="false"
        className="filter_active"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button">
        <div
          className="multiSel">
          <span
            title="Masters Degree">
            Masters Degree
          </span>
          <span
            title="Ph.D">
            Ph.D
          </span>
        </div>
      </a>
    </dt>
    <dd>
      <div
        className="multiSelect">
        <fieldset>
          <legend
            className="sr-only">
            Education level:
          </legend>
          <ul
            style={null}>
            <li>
              <input
                checked={false}
                id="zzz_HS"
                name="education"
                onChange={[Function]}
                type="checkbox"
                value="HS" />
              <label
                htmlFor="zzz_HS">
                High School
              </label>
            </li>
            <li>
              <input
                checked={false}
                id="zzz_AA"
                name="education"
                onChange={[Function]}
                type="checkbox"
                value="AA" />
              <label
                htmlFor="zzz_AA">
                Associates
              </label>
            </li>
            <li>
              <input
                checked={false}
                id="zzz_BA"
                name="education"
                onChange={[Function]}
                type="checkbox"
                value="BA" />
              <label
                htmlFor="zzz_BA">
                Bachelors Degree
              </label>
            </li>
            <li>
              <input
                checked={true}
                id="zzz_MA"
                name="education"
                onChange={[Function]}
                type="checkbox"
                value="MA" />
              <label
                htmlFor="zzz_MA">
                Masters Degree
              </label>
            </li>
            <li>
              <input
                checked={true}
                id="zzz_PHD"
                name="education"
                onChange={[Function]}
                type="checkbox"
                value="PHD" />
              <label
                htmlFor="zzz_PHD">
                Ph.D
              </label>
            </li>
          </ul>
        </fieldset>
      </div>
    </dd>
  </dl>
</div>
`;
